From f885123a15093480febd431f0eacade33c9012dc Mon Sep 17 00:00:00 2001
From: Yasufumi Ogawa <ogawa.yasufumi@lab.ntt.co.jp>
Date: Fri, 25 Jan 2019 07:55:14 +0900
Subject: [PATCH] docs: update server command for registering

This udpate is to add descriptions for registering and unregistering
of REST API server in `server` command.

Signed-off-by: Yasufumi Ogawa <ogawa.yasufumi@lab.ntt.co.jp>
---
 docs/guides/commands/common.rst | 40 +++++++++++++++++++++++++++++++++++-----
 1 file changed, 35 insertions(+), 5 deletions(-)

diff --git a/docs/guides/commands/common.rst b/docs/guides/commands/common.rst
index ec3e3f8..ef862c5 100644
--- a/docs/guides/commands/common.rst
+++ b/docs/guides/commands/common.rst
@@ -87,11 +87,14 @@ Execute command of index of history.
 server
 ------
 
-Show a list of SPP REST API servers and switch to control for multiple
-nodes.
+Switch SPP REST API server.
 
-Show all of registered REST API servers. Run ``server list`` or simply
-``server``.
+SPP CLI is able to manage several SPP nodes via REST API servers.
+It is also abaivable to register new one, or unregister.
+
+Show all of registered servers by running ``server list`` or simply
+``server``. Notice that ``*`` means that the first node is under the
+control of SPP CLI.
 
 .. code-block:: console
 
@@ -103,13 +106,40 @@ Show all of registered REST API servers. Run ``server list`` or simply
       1: 192.168.1.101:7777 *
       2: 192.168.1.102:7777
 
-Switch to other server with index number displayed in ``server list``.
+Switch to other server by running ``server`` with index or address displayed
+in the list. Port number can be omitted if it is default ``7777``.
 
 .. code-block:: console
 
+    # Switch to the second node
     spp > server 2
     Switch spp-ctl to "2: 192.168.1.102:7777".
 
+    # Switch to firt one again with address
+    spp > server 192.168.1.101  # no need for port num
+    Switch spp-ctl to "1: 192.168.1.101:7777".
+
+Register new one by using ``add`` command, or unregister by ``del`` command
+with address. For unregistering, node is also specified with index.
+
+.. code-block:: console
+
+    # Register
+    spp > server add 192.168.122.177
+    Registered spp-ctl "192.168.122.177:7777".
+
+    # Unregister second one
+    spp > server del 2  # or 192.168.1.102
+    Unregistered spp-ctl "192.168.1.102:7777".
+
+You cannot unregister node under the control, or switch to other one before.
+
+.. code-block:: console
+
+    spp > server del 1
+    Cannot del server "1" in use!
+
+
 pwd
 ---
 
-- 
2.7.4

